File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int lst=prices[0],ans=0;
        for(auto p:prices)
        {
            lst=min(lst,p);
            ans=max(ans,p-lst);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) - 1
        area = 0
        while left < right:
            height = min(heights[left], heights[right])
            length = right - left
            area = max(area, height * length)
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        return area